compile error on latest nightly: "type annotations required: cannot resolve `usize: std::ops::AddAssign<_>`"

```
$ rustc --version
rustc 1.22.0-nightly (c6884b12d 2017-09-30)
$ cargo build
   Compiling peresil v0.3.0 (https://github.com/shepmaster/peresil#6049cb14)
   Compiling quote v0.3.15
   Compiling unicode-xid v0.0.4
   Compiling synom v0.11.3
   Compiling syn v0.11.10
   Compiling fuzzy-pickles-derive v0.1.0 (file:///home/david/src/fuzzy-pickles/fuzzy-pickles-derive)
   Compiling fuzzy-pickles v0.1.0 (file:///home/david/src/fuzzy-pickles)
error[E0283]: type annotations required: cannot resolve `usize: std::ops::AddAssign<_>`
   --> src/tokenizer.rs:544:17
    |
544 |             idx += ci.take_while(|&c| UnicodeXID::is_xid_continue(c)).map(|c| c.len_utf8()).sum();
    |                 ^^

error: aborting due to previous error

error: Could not compile `fuzzy-pickles`.

To learn more, run the command again with --verbose.
```

--- a/src/tokenizer.rs
+++ b/src/tokenizer.rs
@@ -541,7 +541,7 @@ fn ident_raw<'s>(_pm: &mut Master<'s>, pt: Point<'s>) -> Progress<'s, (&'s str, 
         if UnicodeXID::is_xid_start(c) || c == '_' {
             idx += c.len_utf8();
 
-            idx += ci.take_while(|&c| UnicodeXID::is_xid_continue(c)).map(|c| c.len_utf8()).sum();
+            idx += ci.take_while(|&c| UnicodeXID::is_xid_continue(c)).map(|c| c.len_utf8()).sum::<usize>();
         }
     }
 

